src/test/java/com/google/jenkins/plugins/k8sengine/CredentialsUtilTest.java

```diff
@@ -0,0 +1,135 @@
+package com.google.jenkins.plugins.k8sengine;
+
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.mockito.ArgumentMatchers.any;
+
+import com.cloudbees.plugins.credentials.CredentialsStore;
+import com.cloudbees.plugins.credentials.SystemCredentialsProvider;
+import com.cloudbees.plugins.credentials.domains.Domain;
+import com.cloudbees.plugins.credentials.domains.DomainRequirement;
+import com.google.api.client.auth.oauth2.Credential;
+import com.google.api.client.googleapis.auth.oauth2.GoogleCredential;
+import com.google.common.collect.ImmutableList;
+import com.google.jenkins.plugins.credentials.oauth.GoogleRobotCredentials;
+import com.google.jenkins.plugins.k8sengine.client.ContainerScopeRequirement;
+import hudson.AbortException;
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+import jenkins.model.Jenkins;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.jvnet.hudson.test.JenkinsRule;
+import org.mockito.Mockito;
+import org.mockito.junit.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class CredentialsUtilTest {
+  private static final String TEST_CREDENTIALS_ID = "test-credentials-id";
+  private static final String TEST_INVALID_CREDENTIALS_ID = "test-invalid-credentials-id";
+
+  @ClassRule public static JenkinsRule r = new JenkinsRule();
+
+  public static Jenkins jenkins;
+
+  @BeforeClass
+  public static void init() throws IOException {
+    jenkins = r.jenkins;
+
+    CredentialsStore store = new SystemCredentialsProvider.ProviderImpl().getStore(jenkins);
+    GoogleRobotCredentials credentials = Mockito.mock(GoogleRobotCredentials.class);
+    credentials.getId();
+    Mockito.when(credentials.getId()).thenReturn(TEST_CREDENTIALS_ID);
+    store.addCredentials(Domain.global(), credentials);
+  }
+
+  @Test(expected = AbortException.class)
+  public void testGetRobotCredentialsReturnsNull() throws AbortException {
+    GoogleRobotCredentials robotCreds =
+        CredentialsUtil.getRobotCredentials(
+            jenkins.get(), ImmutableList.<DomainRequirement>of(), TEST_INVALID_CREDENTIALS_ID);
+    assertNull(robotCreds);
```

Comment:
This is unreachable because the exception is always thrown with null credentials in `getRobotCredentials`, so I think this test can be safely removed as a duplicate with `testGetRobotCredentialsInvalidCredentialsIdAbortException`.

Reply:
Done.